plugin/forward: leaking (*time.Ticker) in goroutine inside (*Transport).connManager

Reported by [Static Major from Orijtech Inc](https://github.com/marketplace/actions/staticmajor-analyzer), we can see this 
```shell
/coredns/plugin/forward/persistent.go:80:4: leaking resource created on line 47
```

which originates from https://github.com/coredns/coredns/blob/2f24b512d3999e9687ffbfaf8f8226c64185dca1/plugin/forward/persistent.go#L47 for which there is a missing `defer ticker.Stop()`

Kindly FYI-ing @willpoint @jhusdero @elias-orijtech @kirbyquerby

--- a/plugin/forward/persistent.go
+++ b/plugin/forward/persistent.go
@@ -45,6 +45,7 @@ func newTransport(addr string) *Transport {
 // connManagers manages the persistent connection cache for UDP and TCP.
 func (t *Transport) connManager() {
 	ticker := time.NewTicker(defaultExpire)
+	defer ticker.Stop()
 Wait:
 	for {
 		select {

